core/src/test/java/org/frankframework/pipes/GetPrincipalPipeTest.java

```diff
@@ -0,0 +1,230 @@
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.pipes;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.ISecurityHandler;
+import org.frankframework.core.PipeForward;
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.core.PipeRunException;
+import org.frankframework.core.PipeRunResult;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.security.Principal;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class GetPrincipalPipeTest extends PipeTestBase<GetPrincipalPipe> {
+	private final String PRINCIPAL_NAME = "TST9";
+	private final String NOT_FOUND_FORWARD_NAME = "notFound";
+	private final String NOT_FOUND_FORWARD_PATH = "doSomething";
+	private ISecurityHandler securityHandler;
+	private Principal principal = mock(Principal.class);
+
+	@BeforeEach
+	public void populateSession() {
+		principal = mock(Principal.class);
+		securityHandler = mock(ISecurityHandler.class);
+		session.put(PipeLineSession.SECURITY_HANDLER_KEY, securityHandler);
+	}
+
+	@Override
+	public GetPrincipalPipe createPipe() throws ConfigurationException {
+		return new GetPrincipalPipe();
+	}
+
+	@Test
+	void missingNotFoundForwardConfigure(){
+		// Given
+		pipe.setNotFoundForwardName(NOT_FOUND_FORWARD_NAME);
+
+		// Expect
+		assertThrows(ConfigurationException.class, () ->
+			// When
+			pipe.configure()
+		, "notInRoleForwardName [notFound] not found");
+	}
+
+	@Test
+	void notFoundForwardConfigure() throws ConfigurationException {
+		// Given
+		PipeForward notFound = new PipeForward(NOT_FOUND_FORWARD_NAME, NOT_FOUND_FORWARD_PATH);
+		pipe.setNotFoundForwardName(NOT_FOUND_FORWARD_NAME);
+		pipe.registerForward(notFound);
+
+		// Expect
+		assertDoesNotThrow(() -> {
+			// When
+			pipe.configure();
+		});
+	}
+
+	@Test
+	void getPrincipalDoPipe() throws Exception {
+		// Given
+		when(principal.getName()).thenReturn(PRINCIPAL_NAME);
+		when(securityHandler.getPrincipal()).thenReturn(principal);
+
+		pipe.configure();
+		pipe.start();
+
+		// When
+		PipeRunResult prr = doPipe(pipe, "", session);
+
+		// Expect
+		String result=prr.getResult().asString();
+		assertEquals("success", prr.getPipeForward().getName());
+		assertEquals(PRINCIPAL_NAME, result);
+	}
+
+	@Test
+	void getNullPrincipalNameDoPipe() throws Exception {
+		// Given
+		when(principal.getName()).thenReturn(null);
+				when(securityHandler.getPrincipal()).thenReturn(principal);
```

Comment:
Please fix your indentation 😉  except if you confess that you're using Notepad.

Reply:
I'm a fan of notepad++ for anything except writing code.

Not sure when and how this happened, will solve this.